lib/codecs/src/encoding/format/gelf.rs

```diff
@@ -322,6 +338,47 @@ mod tests {
         }
     }
 
+    #[test]
+    fn gelf_serializing_timestamp() {
+        // floating point in case of sub second timestamp
+        {
+            let dt = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(61, 1_000_000), Utc);
+
+            let event_fields = btreemap! {
+                VERSION => "1.1",
+                SHORT_MESSAGE => "Some message",
+                HOST => "example.org",
+                TIMESTAMP => dt,
+            };
+
+            let jsn = do_serialize(true, event_fields).unwrap();
+            assert_eq!(true, jsn.get(TIMESTAMP).unwrap().is_f64());
+            assert_eq!(
+                jsn.get(TIMESTAMP).unwrap().as_f64().unwrap(),
+                dt.timestamp_millis() as f64 / 1000.0
+            );
+        }
+
+        // integer in case of no sub second timestamp
+        {
+            let dt = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(61, 0), Utc);
+
+            let event_fields = btreemap! {
+                VERSION => "1.1",
+                SHORT_MESSAGE => "Some message",
+                HOST => "example.org",
+                TIMESTAMP => dt,
+            };
+
+            let jsn = do_serialize(true, event_fields).unwrap();
+            assert_eq!(true, jsn.get(TIMESTAMP).unwrap().is_i64());
+            assert_eq!(
+                jsn.get(TIMESTAMP).unwrap().as_f64().unwrap(),
+                (dt.timestamp_millis() as f64 / 1000.0)
```

Comment:
Seems like we should be getting that as an integer and expecting an integer as well.

Reply:
For this one I would argue that getting as int could potentially be a hard cut off of decimal places (which I have not tested). But I agree that since there is an explicit check if the value is an int beforehand it makes this thinking way overengineered.